Source/Engine/PackageBuilder/SyntaxParser.cs

```diff
@@ -851,6 +905,9 @@ private Syntax ParsePatternReference(string patternName)
                     else
                         throw SyntaxError(TextResource.AttributesAreNotAllowedForStandardPattern, patternName);
                 }
+                else if (pattern.Body is TokenSyntax token)
+                    result = new TokenSyntax(token.TokenKind, token.Text, token.IsCaseSensitive, token.TextIsPrefix,
+                        token.TokenAttributes);
                 else
                     result = pattern.Body;
```

Comment:
This is still incorrect as the body could be a VariationSyntax, see WordBreakPattern in StandardPattern.cs.

I would recommend to create a clone-constructor for Syntax node that has a TextRange argument.